### ISSUE TITLE
New rule: check string literals against regular expressions

### ISSUE THREAD

**Issue by very-amused** [mgechev/revive#510]
In some Go programs, string literals will be primarily used for messages that will face users (such as error messages sent from an API). In these cases, it makes sense to have the ability to enforce a specific kind of formatting for these string literals, such as forced (un)capitalization, forcing or forbidding the usage of certain punctuation, etc. This enforcement can be achieved with the `string-regex` rule.

The rule can be configured with multiple regular expressions, along with custom error messages for each expression (optional). String literals are matched against each expression, creating failures for each expression a string *doesn't* match with.

Example configuration:

```toml
[rule.string-regex]
  arguments = [
    ["/^([^A-Z]|$)/", "must not start with a capital letter"],
    ["/(^|[^\\.!?])$/", "must not end in punctuation"],
    ["/^[^\\n]*$/", "must not contain line breaks"]]
```

**Comment by chavacava** [comment]
Hi, thanks for filling the issue (and the PR)
I've been confronted with this kind of problems with string literals (e.g. with error messages) and I've finally decided to use custom `revive` rules because a generic rule like `rule.string-regex` will generate a lot of _false positives_. For example, running the proposed rule with the example configuration on the [golang/tools](https://github.com/golang/tools.git) or [HUGO](https://github.com/gohugoio/hugo.git) repositories produce a massive number of false positives)

The problem is that string literals are ubiquitous in source code thus imposing constraints on them requires some kind of contextualization to avoid false positives. For example we could imagine to check only the string literals inside calls to `fmt.Errorf` and the like. So, how we could evolve this rule to make it more precise, in other words, to make it produce results with higher confidence?

**Comment by very-amused** [comment]
I think scoping rules to specific method calls like you said is a really good way to accomplish this. I could add a method/function name as the first value in each argument, so `["/^([^A-Z]|$)/", "must not start with a capital letter"]` would become `["fmt.Errorf", "/^([^A-Z]|$)/", "must not start with a capital letter"]`. If this is an acceptable solution to make the rule more precise, I can implement it this weekend.

**Comment by chavacava** [comment]
We can try your proposition and see if the rule is useful (easily configurable, spot problems with high confidence, and avoid false positives) in real cases.

Two details:
1) because the rule will check string literals only in function calls, a renaming of the rule might be necessary
2) the configuration could accept a slice of function names (for example `[["fmt.Errorf","errors.New"], ...]`)
